src/test/java/gameframework/game/GameUniverseDefaultTest.java

```diff
@@ -0,0 +1,126 @@
+package gameframework.game;
+
+import static org.junit.Assert.*;
+
+import gameframework.motion.blocking.MoveBlocker;
+import gameframework.motion.overlapping.Overlappable;
+
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Rectangle;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class GameUniverseDefaultTest {
+
+	protected GameUniverseDefaultImplBis game;
+	protected int cptGameEntityMoveBlocker;
+	protected int cptGameEntityOverlap;
+
+	@Before
+	public void init() {
+		this.game = new GameUniverseDefaultImplBis();
+		this.cptGameEntityMoveBlocker = 0;
+		this.cptGameEntityOverlap = 0;
+		this.game.setGameData(new GameData(new GameConfiguration()));
+	}
+
+	@Test
+	public void testAddGameEntity() {
+		this.game.addGameEntity(createGameEntityMoveBlocker());
+		assertSame(1, this.cptGameEntityMoveBlocker);
+		assertSame(0, this.cptGameEntityOverlap);
+		this.game.addGameEntity(createGameEntityOverlappable());
+		assertSame(1, this.cptGameEntityMoveBlocker);
+		assertSame(1, this.cptGameEntityOverlap);
+	}
+
+	@Test
+	public void testRemoveGameEntity() {
+		GameEntityMoveBlocker mb = createGameEntityMoveBlocker();
+		this.game.addGameEntity(mb);
+		assertSame(1, this.cptGameEntityMoveBlocker);
+		assertSame(0, this.cptGameEntityOverlap);
+		this.game.addGameEntity(createGameEntityOverlappable());
+		assertSame(1, this.cptGameEntityMoveBlocker);
+		assertSame(1, this.cptGameEntityOverlap);
+		this.game.removeGameEntity(mb);
+		assertSame(0, this.cptGameEntityMoveBlocker);
+		assertSame(1, this.cptGameEntityOverlap);
+	}
+	
+	protected GameEntityOverlappable createGameEntityOverlappable() {
+		return new GameEntityOverlappable();
+	}
+	
+	protected GameEntityMoveBlocker createGameEntityMoveBlocker() {
+		return new GameEntityMoveBlocker();
+	}
+	
+	public class GameEntityMoveBlocker implements GameEntity, MoveBlocker{
+		
+		@Override
+		public Rectangle getBoundingBox() {
+			return null;
+		}
+
+		@Override
+		public void draw(Graphics g) {}
+
+		@Override
+		public boolean isMovable() {
+			return false;
+		}
+	}
+	
+	public class GameEntityOverlappable implements GameEntity, Overlappable{
+		
+		@Override
+		public Rectangle getBoundingBox() {
+			return null;
+		}
+
+		@Override
+		public void draw(Graphics g) {}
+
+		@Override
+		public boolean isMovable() {
+			return false;
+		}
+
+		@Override
+		public Point getPosition() {
+			return null;
+		}
+	}
+	
+	public class GameUniverseDefaultImplBis extends GameUniverseDefaultImpl {
```

Comment:
that's not acceptable. You duplicate the code of the superclass in your test. What happens if the superclass changes?